pkg/virt-handler/vm.go

```diff
@@ -1848,20 +1818,16 @@ func (d *VirtualMachineController) closeLauncherClient(vmi *v1.VirtualMachineIns
 		return nil
 	}
 
-	// maps require locks for concurrent access
-	d.launcherClientLock.Lock()
-	defer d.launcherClientLock.Unlock()
-
-	clientInfo, ok := d.launcherClients[vmi.UID]
-	if ok && clientInfo.client != nil {
-		if clientInfo.client != nil {
-			clientInfo.client.Close()
-			close(clientInfo.domainPipeStopChan)
+	clientInfo, loaded := d.launcherClients.LoadAndDelete(vmi.UID)
```

Comment:
I don't think that we can do the get and delete at the same time. We do some cleanup which has to be done. IF we would have an error with e.g. `os.RemoveAll` we would not clean up the ghost record. I think you have to get here and do the delete where it was done before.

Reply:
Going back to Load/Delete that the original code, nice catch btw.